test_util/test_upgrade_vpc.py

```diff
@@ -353,7 +353,11 @@ def marathon_app_tasks_survive_upgrade():
             def test_mesos_task_state_remains_consistent():
                 # Verify that the "state" of the task does not change.
                 task_state_end = self.get_master_task_state(dcos_api, self.tasks_start[self.test_app_ids[0]][0])
-                if not self.task_state_start == task_state_end:
+                # To avoid errors when new items are added to the task state, we
+                # check that the old state is a subset of the new state.
+                if (task_state_end is None or
+                        self.task_state_start is None or
+                        not all(item in task_state_end.items() for item in self.task_state_start.items())):
```

Comment:
@mellenburg see the comment above for why I'm changing that clause - the Mesos update in this PR adds a new field to the task state. If we check that the old state is a subset of the new state, rather than for strict equality, then we can avoid breaking this test when new fields are added. WDYT?